js/url_search_params.ts

```diff
@@ -106,8 +112,30 @@ export class URLSearchParams {
    *       searchParams.set('name', 'value');
    */
   set(name: string, value: string): void {
-    this.delete(name);
-    this.append(name, value);
+    // If there are any name-value pairs whose name is name, in list,
+    // set the value of the first such name-value pair to value
+    // and remove the others.
+    let found = false;
+    let i = 0;
+    while (i < this.params.length) {
```

Comment:
This could be written like:

```ts
while (i < this.params.length) {
  if (this.params[i][0] === name) {
    if (!found) {
      this.params[i][1] = value;
      found = true;
    } else {
      this.params.splice(i, 1);
      continue;
    }
  }
  i++;
}
```